netbox/utilities/forms/fields.py

```diff
@@ -150,6 +153,72 @@ def to_python(self, value):
         return value
 
 
+class MemoryField(forms.MultiValueField):
+    widget = widgets.MemoryWidget
+    MULTIPLIERS = {
+        MemoryUnitChoices.UNIT_MB: 1024**0,
+        MemoryUnitChoices.UNIT_GB: 1024**1,
+        MemoryUnitChoices.UNIT_TB: 1024**2,
+    }
+
+    def __init__(self, **kwargs):
+        print(kwargs)
+        fields = (
+            forms.IntegerField(required=False),
+            forms.ChoiceField(
+                choices=MemoryUnitChoices.MEMORY_CHOICES,
+                required=False
+            ),
+        )
+        super(MemoryField, self).__init__(
+            fields=fields, required=False,
+            require_all_fields=False, **kwargs
+        )
+
+    @classmethod
+    def compress(cls, data):
+        if data:
+            size, unit = data
+            if size and not unit:
+                raise forms.ValidationError("Memory unit cannot be blank.")
+            elif size and unit:
+                return size * cls.MULTIPLIERS[unit]
+            elif not size and unit:
+                raise forms.ValidationError("Please enter a memory value when unit is selected.")
+
+
+class DiskField(forms.MultiValueField):
```

Comment:
This is essentially a duplicate of `MemoryField`; both the memory and disk fields should use the same form field class. Instead of declaring `MULTIPLIERS` as a constant, you could accept a list of units as a kwarg on init.

Reply:
I'm having trouble passing the MULTIPLIERS through to the compress function as the compress function class is based on django MultiValueField not MemoryField. Need to work on a solution to get around this.